examples/generate_bom/README.md

```diff
@@ -42,22 +46,96 @@ options:
+
+This script relies on the presence of an `attributes_mapping.json` file next to
+it. This file maps the Component Attributes in the SchDoc files to the columns
+of the BOM. The example version of `attributes_mapping.json` is:
+
+```json
+{
+    "description": ["PART DESCRIPTION"],
+    "designator": ["Designator"],
+    "manufacturer": ["Manufacturer", "MANUFACTURER"],
+    "part_number": ["PART", "MANUFACTURER #"]
+}
+```
+
+In this file, the keys are the names of the columns in the BOM, and the values
+are a list of the names of the attributes in the SchDoc files that should be
+mapped to that column. For example, if your part number is stored either in the
+`PART` or `MANUFACTURER #` attribute, you would add both of those to the list.
+If there is only one attribute, you can omit the list and just use a string. The
+script checks these attributes in order, and uses the _first_ one it finds. So
+if both `PART` and `MANUFACTURER #` are defined, it will use `PART`.
+
+If you need to customize the attributes further or have more complex logic,
+customize the `map_json_to_component` function in the script. Note that the
+AllSpice generated JSON may change at any time, so everything except the
+component attributes may change!
+
 You can also import this python file in another file to use the methods defined
 in it. If you want the BOM in a different format, or have other requirements,
 you can read and adapt this code, or use it as a reference for your own code.
+
+## COGS Calculation
+
+The COGS script depends on the BOM script. It takes a BOM CSV file and computes
+the COGS at various quantities of PCBs made. The COGS is computed by looking up 
+the part number using an API you define in the script via the
+`fetch_price_for_part` method. An example is implemented using the cofactr API.
+
+To use the script, start by running:
+
+```bash
+python3 compute_cogs.py -h
+```
+
+Which prints the following help text:
+
+```
+usage: compute_cogs.py [-h] [--quantities QUANTITIES]
+                       [--output-file OUTPUT_FILE]
+                       bom_file
+
+positional arguments:
+  bom_file              The path to the BOM file.
+
+options:
+  -h, --help            show this help message and exit
+  --quantities QUANTITIES
+                        A comma-separated list of quantities of PCBs to
+                        compute the COGS for. E.g. 1,10,100,1000. Defaults to
+                        the example.
+  --output-file OUTPUT_FILE
+                        The path to the output file. Defaults to stdout, i.e.
+                        printing to the console.
+```
+
+If you have generated a BOM using `generate_bom.py`, you should first take a
+look at the bom and remove rows for testpoints and other parts which don't need
+to be in the COGS computation. This step is not strictly necessary, but will
+make the COGS computation faster and more accurate, as the price lookup may get
+confused by testpoints and other parts.
```

Comment:
What would confuse it?  If it can't find something?  If that's what the problem is, it might be nice to say that explicitly.  Otherwise, it scares me into thinking this script isn't reliable.